lib/inspec/metadata.rb

```diff
@@ -52,57 +53,23 @@ def supports(sth, version = nil)
       # already.
     end
 
-    def is_supported?(os, entry)
-      name = entry[:'os-name'] || entry[:os]
-      family = entry[:'os-family']
-      release = entry[:release]
-
-      # return true if the backend matches the supported OS's
-      # fields act as masks, i.e. any value configured for os-name, os-family,
-      # or release must be met by the backend; any field that is nil acts as
-      # a glob expression i.e. is true
-
-      # os name is both saved in :family and :name, so check both
-      name_ok = name.nil? ||
-                os[:name] == name || os[:family] == name
-
-      family_check = family.to_s + '?'
-      family_ok = family.nil? || os[:family] == family ||
-                  (
-                    os.respond_to?(family_check) &&
-                    # this call will return true if the family matches
-                    os.method(family_check).call
-                  )
-
-      # ensure we do have a string if we have a non-nil value eg. 16.06
-      release_ok = release.nil? || os[:release] == release
-
-      # we want to make sure that all matchers are true
-      name_ok && family_ok && release_ok
-    end
-
     def inspec_requirement
-      inspec = params[:supports].find { |x| !x[:inspec].nil? } || {}
-      Gem::Requirement.create(inspec[:inspec])
+      inspec = params[:supports].find { |x| !x[:inspec].nil? }
+      if inspec
+        Inspec::Log.warn '[DEPRECATED] The use of inspec.yml `supports:inspec` is being deprecated in InSpec 2.0. Please use `inspec_version` instead.'
+        Gem::Requirement.create(inspec[:inspec])
+      else
+        Gem::Requirement.create(params[:inspec_version])
```

Comment:
It might be worth a comment here for the next reader who is not familiar with `Gem::Requirement.create` that if you pass in `nil`, it will create a constraint of `[">= 0"]` so it's totally OK for this to be `nil`.